File: add_files_to_xcode.rb
#!/usr/bin/env ruby
require 'xcodeproj'

project_path = '/Users/alwan/FieldMind/fieldmind/fieldmind.xcodeproj'
project = Xcodeproj::Project.open(project_path)

# 获取主 target
target = project.targets.first

# 核心文件列表
core_files = [
  'fieldmind/Core/UnifiedAppState.swift',
  'fieldmind/Core/DataPipeline/DataPipeline.swift',
  'fieldmind/Core/DataPipeline/CleanDataPipeline.swift',
  'fieldmind/Core/DataPipeline/RawDataPipeline.swift',
  'fieldmind/Services/BackendService.swift',
  'fieldmind/Services/KnowledgeVaultService.swift',
  'fieldmind/Services/DistillationService.swift',
  'fieldmind/Views/MainNavigationView.swift',
  'fieldmind/Views/KnowledgeVaultView.swift',
  'fieldmind/Views/DistillationView.swift'
]

# 获取或创建组
def get_or_create_group(project, path_components)
  current_group = project.main_group
  path_components.each do |component|
    found = current_group.children.find { |child| child.display_name == component && child.isa == 'PBXGroup' }
    if found
      current_group = found
    else
      current_group = current_group.new_group(component)
    end
  end
  current_group
end

added_count = 0
skipped_count = 0

core_files.each do |file_path|
  full_path = File.join('/Users/alwan/FieldMind', file_path)

  unless File.exist?(full_path)
    puts "⚠️  文件不存在: #{file_path}"
    skipped_count += 1
    next
  end

  # 检查文件是否已在项目中
  existing_file = project.files.find { |f| f.path == file_path }
  if existing_file
    puts "⏭️  已存在: #{file_path}"
    skipped_count += 1
    next
  end

  # 解析路径并获取/创建对应的组
  path_parts = file_path.split('/')
  file_name = path_parts.pop
  group_path = path_parts[1..-1] # 跳过 'fieldmind' 前缀

  group = get_or_create_group(project, group_path)

  # 添加文件引用
  file_ref = group.new_file(full_path)

  # 添加到编译阶段
  target.source_build_phase.add_file_reference(file_ref)

  puts "✅ 已添加: #{file_path}"
  added_count += 1
end

# 保存项目
project.save

puts "\n" + "="*60
puts "📊 文件添加统计"
puts "="*60
puts "✅ 成功添加: #{added_count} 个文件"
puts "⏭️  跳过: #{skipped_count} 个文件"
puts "="*60
puts "\n🎉 Xcode 项目已更新！"
puts "📝 下一步: 在 Xcode 中打开项目并编译测试"

File: add_files_to_xcode.py
#!/usr/bin/env python3
"""
自动将核心集成文件添加到 Xcode 项目
使用 pbxproj 库来操作 .xcodeproj 文件
"""

import os
import sys

try:
    from pbxproj import XcodeProject
except ImportError:
    print("正在安装 pbxproj...")
    os.system("pip3 install pbxproj")
    from pbxproj import XcodeProject

# 项目路径
PROJECT_PATH = '/Users/alwan/FieldMind/fieldmind/fieldmind.xcodeproj'
BASE_PATH = '/Users/alwan/FieldMind/fieldmind/fieldmind'

# 核心文件列表（相对于 BASE_PATH）
CORE_FILES = [
    'Core/UnifiedAppState.swift',
    'Core/DataPipeline/DataPipeline.swift',
    'Services/BackendService.swift',
    'Services/KnowledgeVaultService.swift',
    'Views/KnowledgeVaultView.swift',
]

# 这些文件在外层 fieldmind/ 目录
OUTER_FILES = [
    '../Services/DistillationService.swift',
    '../Views/MainNavigationView.swift',
    '../Views/DistillationView.swift',
]

def main():
    print("="*60)
    print("🔧 开始添加文件到 Xcode 项目")
    print("="*60)

    # 打开项目
    try:
        project = XcodeProject.load(PROJECT_PATH + '/project.pbxproj')
    except Exception as e:
        print(f"❌ 无法打开项目: {e}")
        return 1

    added_count = 0
    skipped_count = 0
    missing_count = 0

    # 处理所有文件
    all_files = CORE_FILES + OUTER_FILES

    for file_path in all_files:
        full_path = os.path.join(BASE_PATH, file_path)
        # 标准化路径
        full_path = os.path.normpath(full_path)

        # 用于显示的相对路径
        display_path = os.path.relpath(full_path, '/Users/alwan/FieldMind/fieldmind')

        # 检查文件是否存在
        if not os.path.exists(full_path):
            print(f"⚠️  文件不存在: {display_path}")
            missing_count += 1
            continue

        # 检查是否已在项目中（使用文件名检查）
        file_name = os.path.basename(file_path)
        if project.get_files_by_name(file_name):
            print(f"⏭️  已存在: {display_path}")
            skipped_count += 1
            continue

        # 添加文件
        try:
            # 获取文件的相对目录
            rel_dir = os.path.dirname(display_path)
            project.add_file(full_path, parent=project.get_or_create_group(rel_dir) if rel_dir else None)
            print(f"✅ 已添加: {display_path}")
            added_count += 1
        except Exception as e:
            print(f"❌ 添加失败 {display_path}: {e}")
            continue

    # 保存项目
    if added_count > 0:
        try:
            project.save()
            print("\n💾 项目已保存")
        except Exception as e:
            print(f"\n❌ 保存失败: {e}")
            return 1

    # 统计
    print("\n" + "="*60)
    print("📊 文件添加统计")
    print("="*60)
    print(f"✅ 成功添加: {added_count} 个文件")
    print(f"⏭️  已存在跳过: {skipped_count} 个文件")
    print(f"⚠️  文件缺失: {missing_count} 个文件")
    print("="*60)

    if added_count > 0:
        print("\n🎉 Xcode 项目已更新！")
        print("📝 下一步:")
        print("   1. 在 Xcode 中打开项目: open fieldmind/fieldmind.xcodeproj")
        print("   2. 编译项目: ⌘+B")
        print("   3. 运行测试: ⌘+R")

    return 0

if __name__ == '__main__':
    sys.exit(main())
